File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.java
class Solution {
    public int solution(int[][] triangle) {
        int i;
        int size;
        int[] dp;

        size = triangle.length;
        dp = new int[size];
        System.arraycopy(triangle[size - 1], 0, dp, 0, size);
        while (--size > 0) {
            for (i = 0; i < size; i++) {
                dp[i] = Math.max(dp[i], dp[i + 1]) + triangle[size - 1][i];
            }
        }
        return dp[0];
    }
}

File: 프로그래머스/3/43105. 정수 삼각형/정수 삼각형.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> triangle) {
    int i;
    int j;
    int len;

    len = triangle.size();
    for (i = len - 2; i >= 0; i--) {
        for (j = 0; j <= i; j++) {
            triangle[i][j] += max(triangle[i + 1][j], triangle[i + 1][j + 1]);
        }
    }
    return triangle[0][0];
}